roachtest: skip and reown tests

tbg opened the pull request:
cc @petermattis, let me know if you're opposed to owning the disk-stalled roachtests. I saw while triaging that you and Bilal were discussing fixing the charybdefs setup and I also saw talk about disk stall work in the weekly notes.

Release justification: testing only
Release note: None

tbg committed:
roachtest: own disk-stalled roachtest to storage team

Release justification: testing-only
Release note: None

tbg committed:
roachtest: require 20.1+ for some tpccbench versions

Don't run `tpccbench/nodes=9/cpu=4/`chaos/partition,/multi-region}` no
19.2 and older. These tests are hard to stabilize and we are unlikely to
have the bandwidth to do so on old release branches.

Release justification: testing change
Release note: None

tbg committed:
roachtest: run jepsen on 20.1+ only

It generates a fair amount of flakes and we're unlikely to land any
changes on these old branches that can expose new behavior.

Release justification: testing
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53544)
<!-- Reviewable:end -->

tbg committed:
roachtest: de-flake gossip/peerings

Pulling the gossip info via HTTP can fail if the freshly restarted nodes
didn't gossip the actual cluster settings yet. Fixing the initialization
is tracked via #50271.

Closes #48005.

Release justification: testing
Release note: None

petermattis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> for changing the ownership of `disk-stalled` to storage.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53544#-:-MFjrjGb1a57TeJ6ujIu:b-dkixiz)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @nvanbenschoten)



<!-- Sent from Reviewable.io -->

nvb reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53544#-:-MFlASEV-_vi3soM_huk:bnfp4nl)* status: :shipit: complete! 2 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

tbg commented:
bors r=nvanbenschoten,petermattis
